src/plot_hourly_line_graph.py

```diff
@@ -3,32 +3,61 @@
 import datetime
 import csv
 import numpy
+import os
+
+directory = r"..\graphs\hourly_line_graph"
+if not os.path.exists(directory):
+    os.makedirs(directory)
 
 list = []
-x = []
-y = []
-bins_x = []
 
 with open('../data/data.csv', 'r') as csvfile:
-    plots = csv.reader((l.replace('\0', '') for l in csvfile))
+    plots = csv.reader(csvfile, delimiter=',')
     for row in plots:
         try:
             list.append([datetime.datetime.strptime(
-                row[0], '%m/%d/%y %H:%M'), int(row[1])])
+                row[0], '%Y-%m-%d %H:%M:%S.%f'), int(row[1])])
         except:
             pass
 
-bins_x = []
-bins_y = []
+bins_x = set()
+date_dict = {}
+start_count = None
+start_date = None
 
 for element in list:
```

Comment:
I found this part very hard to read. It might be better if we create a class for a single entry of the count data.

Something like:

```
class CountData:
    def __init__(self, timestamp, count):
        self.timestamp = timestamp
        self.count = count
```

So then instead of having to create key value pairs of **date:tuple of lists**, our key value pairs can be **date: CountData object**

More readable:
```
for timestamp, count in data:
    if not timestamp.date() in sensor-reading_day_map:
        sensor-reading_day_map[timestamp.date()] = []
        current_date = timestamp.date()
        date_start_count = count
    sensor-reading_day_map[timestamp.date()].append(CountData(timestamp, count - date_start_count))
```

(I haven't tested this code so idk if it'll work but something like that)